ibacm/src/acm.c

```diff
@@ -2596,17 +2596,64 @@ acm_open_port(struct acmc_port *port, struct acmc_device *dev, uint8_t port_num)
 	port->state = IBV_PORT_DOWN;
 }
 
+static bool dev_support_ib(struct ibv_device *ibdev)
+{
+	struct ibv_context *verbs;
+	struct ibv_device_attr attr;
+	struct ibv_port_attr port_attr;
+	int i, ret;
+	bool has_ib_port = false;
+
+	verbs = ibv_open_device(ibdev);
+	if (verbs == NULL) {
+		acm_log(0, "ERROR - opening device %s\n", ibdev->name);
+		return false;
+	}
+
+	memset(&attr, 0, sizeof(attr));
+	ret = ibv_query_device(verbs, &attr);
+	if (ret) {
+		acm_log(0, "ERROR - ibv_query_device (%d) %s\n", ret, ibdev->name);
+		return false;
+	}
+
+	for (i = 0; i < attr.phys_port_cnt; i++) {
+		memset(&port_attr, 0, sizeof(port_attr));
+
+		ret = ibv_query_port(verbs, i + 1, &port_attr);
+		if (ret) {
+			acm_log(0, "ERROR - ibv_query_port (%d)\n", ret);
+			continue;
+		}
+		if (port_attr.link_layer == IBV_LINK_LAYER_INFINIBAND) {
+			acm_log(1, "%s port %d is an InfiniBand port\n", ibdev->name, i);
+			has_ib_port = true;
+		} else
+			acm_log(1, "%s port %d is not an InfiniBand port\n", ibdev->name, i);
+	}
+
+	ibv_close_device(verbs);
+
+	return has_ib_port;
+}
+
 static void acm_open_dev(struct ibv_device *ibdev)
 {
 	struct acmc_device *dev;
 	struct ibv_device_attr attr;
-	struct ibv_port_attr port_attr;
 	struct ibv_context *verbs;
 	size_t size;
 	int i, ret;
-	unsigned int opened_ib_port_cnt = 0;
+	bool has_ib_port;
 
 	acm_log(1, "%s\n", ibdev->name);
+
+	has_ib_port = dev_support_ib(ibdev);
+	if (!has_ib_port) {
```

Comment:
I guess you do not need this local bool has_ib_port? Why not 

if (dev_support_ib(ibdev)) {